src/javascript/EditPanel/EditPanelContent/FormBuilder/FieldSet/Field/Field.jsx

```diff
@@ -77,17 +79,22 @@ export const FieldCmp = ({classes, inputContext, idInput, selectorType, field, f
     const wipInfo = values[Constants.wip.fieldName];
 
     // Lookup for registered on changes for given field selector type
-    const registeredOnChanges = [...registry.find({
+    const registeredOnChanges = useMemo(() => [...registry.find({
         type: 'selectorType.onChange',
         target: selectorType.key
-    }), ...registry.find({type: 'selectorType.onChange', target: '*'})];
-    const registeredOnChange = currentValue => {
+    }), ...registry.find({type: 'selectorType.onChange', target: '*'})], [selectorType.key]);
+
+    const onChangeContext = useRef({});
+    useEffect(() => {
+        onChangeContext.current = {...editorContext, ...sectionsContext, formik, client};
+    }, [editorContext, sectionsContext, formik, client]);
```

Comment:
Use a ref for changeContext to avoid adding dependencies on formik,editorContext, sectionContext to registeredOnChange callback. Ref value is used only when calling onChange callback.